File: socketio_rooms_flutter/lib/models/user_model.dart
import 'dart:convert';

class UserModel {
  final String idUser;

  UserModel({
    required this.idUser,
  });

  Map<String, dynamic> toMap() {
    return {
      "idUser": idUser,
    };
  }

  // Convertir el objeto Message a JSON (cadena de texto) Para enviar en websocket
  String toJson() {
    return jsonEncode(toMap());
  }

  factory UserModel.fromMap(Map<String, dynamic> map) {
    return UserModel(
      idUser: map["idUser"] ?? "default",
    );
  }

  // Crear un Message a partir de JSON (cadena de texto) Para recibir en websocket
  factory UserModel.fromJson(String source) {
    return UserModel.fromMap(jsonDecode(source));
  }
}

File: socketio_server_nodejs/src/models/UserModel.js
class UserModel {
    constructor(idUser) {
        this.idUser = idUser;
    }

    toMap() {
        return {
            idUser: this.idUser,
        };
    }

    toJson() {
        return JSON.stringify(this.toMap()); // Serializar el mapa a JSON
    }

    static fromMap(map) {
        return new UserModel(map.idUser);
    }

    static fromJson(source) {
        return UserModel.fromMap(JSON.parse(source)); // Deserializar el JSON
    }

}

module.exports = UserModel;